wait/host.go

```diff
@@ -586,43 +586,60 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is a Notification available with the given name and the optional conditions
+func (a *HostAwaitility) WaitForNotifications(name, reason string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notifications []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: name, toolchainv1alpha1.NotificationTypeLabelKey: reason}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of notification '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		for _, n := range notificationList.Items {
+			for _, match := range criteria {
+				if match(a, &n) {
+					a.T.Logf("found notification '%s'", name)
+					notifications = append(notifications, &n)
+				}
 			}
 		}
-		a.T.Logf("found notification '%s'", name)
-		notification = obj
+
+		if len(notifications) <= 0 {
+			return false, nil
+		}
 		return true, nil
 	})
-	return notification, err
+	return notifications, err
 }
 
 // WaitUntilNotificationDeleted waits until the Notification with the given name is deleted (ie, not found)
-func (a *HostAwaitility) WaitUntilNotificationDeleted(name string) error {
+func (a *HostAwaitility) WaitUntilNotificationDeleted(name, reason string) error {
 	return wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		notification := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, notification); err != nil {
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: name, toolchainv1alpha1.NotificationTypeLabelKey: reason}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("Notification has been deleted '%s'", name)
 				return true, nil
 			}
 			return false, err
 		}
-		a.T.Logf("waiting until Notification is deleted '%s'", name)
-		return false, nil
+
+		if len(notificationList.Items) > 0 {
+			a.T.Logf("waiting until Notification is deleted '%s'", notificationList.Items[0].Name)
```

Comment:
I would rather print all of them, not only the first one

Reply:
Fixed here: 258e10b3c4259d21be1b08d46f8b59ed0cced242 and as part of: 72d3c5c66de1c858dfee03ed84878e0fe86996de